### PR TITLE
Update readme

### LINKED ISSUE #118: README is outdated
README mentions dep instead of go modules.
The list of CRDs for streaming is incomplete
The set of additional dependencies is outdated (knative build ...)

### PR DESCRIPTION
**Pull request by scothis**
Resolves #118

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,72 +15,75 @@ riff System contains three API groups with CustomResourceDefinitions that are th
...
+- `riff-${component}-readonly-role` - read access to component's CRDs
 
 These roles are aggregated to the `edit` and `view` ClusterRoles respectively.
 
 See the Kuberneties [Using RBAC Authorization](https://kubernetes.io/docs/reference/access-authn-authz/rbac/) for more information.
 
 ## Install
 
-riff System is not installed directly for typical use. See the [Getting Started](https://projectriff.io/docs/getting-started/) docs to learn how to install riff.
+riff System is typical not installed directly by end users. See the [Getting Started](https://projectriff.io/docs/getting-started/) docs to learn how to install riff.
```
Comment (ericbottard) [inline README.md]:
```suggestion
riff System is typically not installed directly by end users. See the [Getting Started](https://projectriff.io/docs/getting-started/) docs to learn how to install riff.
```

File: README.md [hunk 2]
```diff
@@ -15,72 +15,75 @@ riff System contains three API groups with CustomResourceDefinitions that are th
...
+ko apply -f config/riff-${component}.yaml
 ```
 
 Additional dependencies must be installed independently into the cluster including:
 
-- riff build templates
-- Knative Build
-- Knative Serving
-- Istio
+- [Istio](https://istio.io) (knative runtime)
```
Comment (ericbottard) [inline README.md]: Maybe add "needed by" in the parentheses where appropriate, as this may not be immediately clear.

File: README.md [hunk 3]
```diff
@@ -15,72 +15,75 @@ riff System contains three API groups with CustomResourceDefinitions that are th
...
   - `Adapter` - adapters map applications, functions or container images into an existing Knative Service or Configuration.
   - `Deployer` - deployers map HTTP requests to applications, functions, containers or images with Knative
 
-### Runtime
+### Runtimes
 
-A `controller` and `webhook` Deployments exist in the `riff-system` namespace to validate and reconcile the riff CRDs.
+Each runtime has an independent manager deployment that reconciles and hosts validation and defaulting webhooks for the CRDs needed by that runtime.
+
+The build component is required by each runtime.
```
Comment (ericbottard) [inline README.md]: This kinda contradicts the the "independent" sentence above, although the word "runtime" (of which `build` isn't) hasn't been introduced in that README yet, AFAICT

Comment (ericbottard) [reply README.md]: Maybe just state that core, streaming and knative are "runtimes" as the first sentence of that section.

File: README.md [hunk 4]
```diff
@@ -15,72 +15,75 @@ riff System contains three API groups with CustomResourceDefinitions that are th
...
 
 ## Install
 
-riff System is not installed directly for typical use. See the [Getting Started](https://projectriff.io/docs/getting-started/) docs to learn how to install riff.
+riff System is typical not installed directly by end users. See the [Getting Started](https://projectriff.io/docs/getting-started/) docs to learn how to install riff.
 
 ## Development
 
-After making any changes to source files in `./pkgs/apis` it is necessary to regenerate the API client by running:
+riff System builds on [KubeBuilder](https://www.kubebuilder.io) which makes extensive use of generated code and configuration. After making any changes to source files in `./pkg/apis` or `./pkg/controllers` it is necessary to regenerate the API client and configuration by running:
```
Comment (ericbottard) [inline README.md]: After reading the slack discussion, should we also explain that this step is also required after changing kustomize files, and also that files directly under config/ should not be edited directly?

Comment (scothis) [reply README.md]: some config files can (or must) be edited directly. It's worth following up

**Pull request merged**